### PR TITLE
Add a test fixture for pushing various memory limits.

### PR DESCRIPTION
**Pull request by acw**
This fixture, written in WAT to ensure that none of our compilers do
anything too smart, allows the caller to ask it to allocate memory in
various ways: on the wasm heap, in headers, or in the response body. It
also allows the user to ask it to stall for a certain amount of time.
This is handy for testing some edge cases in resource limits.

### CODE REVIEW

File: test-fixtures/combined_heap_limits.wat [hunk 1]
```diff
@@ -0,0 +1,588 @@
...
+      (global.get $request_handle_buffer)
+      (global.get $request_body_buffer)
+      (call $request_get)
+      (call $maybe_error_die)
+    )
+
+  ;; get the size the user wants from the provided header
+  ;; we're going to cheat a bit, here, and assume that the only
+  ;; reason we might get an error is because the user didn't pass
+  ;; this header; and if they didn't pass a header, we're going to
```
Comment (aturon) [inline test-fixtures/combined_heap_limits.wat:345]: this comment seems to trail off

File: test-fixtures/combined_heap_limits.wat [hunk 2]
```diff
@@ -0,0 +1,588 @@
...
+        (br_if $body_write_block) ;; remember, this cancels execution of the
+                                  ;; rest of the block
+
+        ;; OK, we'll just tack on 1k more data
+        (global.get $response_body_buffer)
+        (i32.load) ;; push the handle
+        (i32.const 0) ;; push the pointer for the buffer; the start of memory,
+                      ;; because we don't care
+        (i32.const 1024) ;; the length of the buffer
+        (i32.const 0) ;; I think this means add it to the end?
```
Comment (aturon) [inline test-fixtures/combined_heap_limits.wat:313]: yup

**Review by aturon — DISMISSED** [review]
Incredible work. 💯

File: test-fixtures/combined_heap_limits.wat (hunk 2 above)
Comment (acw) [reply test-fixtures/combined_heap_limits.wat:313]: I figured, but I'll just grump that the assumption that `witx` enumerations are numbered in order from zero is not super clearly stated anywhere I could find.

File: test-fixtures/combined_heap_limits.wat (hunk 1 above)
Comment (acw) [reply test-fixtures/combined_heap_limits.wat]: Heh. Fixed in 6d883f6.

**Review by aturon — APPROVED** [review]
